force-app/main/default/classes/ProgramEngagementSelector.cls

```diff
@@ -15,11 +15,22 @@ public with sharing class ProgramEngagementSelector {
         if (!Schema.SObjectType.ProgramEngagement__c.isAccessible()) {
             return new List<ProgramEngagement__c>();
         }
-        List<ProgramEngagement__c> queriedEngagements = [
+        List<ProgramEngagement__c> queriedEngagements = new List<ProgramEngagement__c>();
+
+        String query = 'SELECT Id, Name, Program__c, Program__r.Name';
```

Comment:
@bmuniswamy  Can you describe what you're doing here?  I'd imagine we could run into trouble if contactId is null and we try to query ALL Program Engagements in a very large org.

Reply:
@AndersonTarren , This will not be required anymore. I did this with the intention that we will show program engagements as picklist if contact does not exist. But since are not showing it as a picklist and will be displaying it as a lookup this can be removed.